src/main/java/com/lmax/simpledsl/api/SimpleDslArg.java

```diff
@@ -107,6 +110,35 @@ public SimpleDslArg setAllowedValues(final String... allowedValues)
         return this;
     }
 
+    /**
+     * Restrict the allowed values for this argument to the specified set.
+     * <p>
+     * Specifying a value outside this set will result in an exception being thrown when parsing the arguments.
+     *
+     * @param <T> the type
+     * @param clazz the {@link Class} that provides the allowed values.
+     * @return this argument
+     * @throws IllegalArgumentException if allowed values cannot be determined from the provided class
+     */
+    public <T> SimpleDslArg setAllowedValues(final Class<T> clazz)
```

Comment:
Not sure this change is mentioned in the PR, and think I might refactor this method later, but it's an improvement on not existing before so :man_shrugging: